### ISSUE TITLE
vcat on SparseVector's of different eltype returns a SparseMatrixCSC on v0.6-rc2

### ISSUE THREAD

**Issue by blegat** [JuliaLang/julia#22225]
I would expect the concatenation of sparse vectors to be a sparse vector but it is not the case when their `eltype` is different
```julia
julia> versioninfo()
Julia Version 0.6.0-rc2.0
Commit 68e911b (2017-05-18 02:31 UTC)

julia> vcat(spzeros(2), spzeros(3)) # Same eltype
5-element SparseVector{Float64,Int64} with 0 stored entries

julia> vcat(spzeros(Int, 2), spzeros(3)) # Different eltype
5×1 SparseMatrixCSC{Float64,Int64} with 0 stored entries
```
This is not a regression since the bug is present in Julia v0.5 too
```julia
julia> versioninfo()
Julia Version 0.5.2
Commit f4c6c9d4bb* (2017-05-06 16:34 UTC)

julia> vcat(spzeros(Int, 2), spzeros(Float64, 3))
5×1 sparse matrix with 0 Float64 nonzero entries
```

**EDIT**: The following bug is however a regression:
```julia
julia> versioninfo()
Julia Version 0.6.0-rc2.0
Commit 68e911b (2017-05-18 02:31 UTC)

julia> vcat(Float64[], spzeros(1))
1×1 SparseMatrixCSC{Float64,Int64} with 0 stored entries
```
```julia
julia> versioninfo()
Julia Version 0.5.2
Commit f4c6c9d4bb* (2017-05-06 16:34 UTC)

julia> vcat(Float64[], spzeros(1))
Sparse vector of length 1 with 0 Float64 nonzero entries:
```

**Comment by ararslan** [comment]
Fixed by #22299 and #22301.

**Comment by blegat** [comment]
Thank you @Sacha0 !
